internal/cloud/realm/import.go

```diff
@@ -0,0 +1,47 @@
+package realm
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/10gen/realm-cli/internal/utils/api"
+)
+
+const (
+	importPathPattern = appPathPattern + "/import"
+
+	importQueryStrategy = "strategy"
+
+	importStrategyReplaceByName = "replace-by-name"
+)
+
+// ImportRequest is a Realm application import request
+type ImportRequest struct {
+	ConfigVersion AppConfigVersion `json:"config_version"`
+	AuthProviders []AuthProvider   `json:"auth_providers"`
+}
+
+func (c *client) Import(groupID, appID string, req ImportRequest) error {
+	if req.ConfigVersion == AppConfigVersionNil {
+		req.ConfigVersion = DefaultAppConfigVersion
+	}
+
+	res, resErr := c.doJSON(
+		http.MethodPost,
+		fmt.Sprintf(importPathPattern, groupID, appID),
+		req,
+		api.RequestOptions{
+			UseAuth: true,
+			Query: map[string]string{
+				importQueryStrategy: importStrategyReplaceByName,
+			},
+		},
+	)
+	if resErr != nil {
+		return resErr
+	}
+	if res.StatusCode != http.StatusNoContent {
+		return UnmarshalServerError(res)
```

Comment:
related, do we need a `defer res.Body.Close()` here?

Reply:
yeah, so actually I put a `defer res.Body.Close()` at the start of `UnmarshalServerError`, b/c I'm not closing the response body anywhere I call this yet

Reply:
that's _probably_ okay in this case, but I'm a bit hesitant to pass off the responsibility of closing a `ReadCloser` to a helper like this. it feels like it should still be the responsibility of the "owner" of that `ReadCloser` to clean things up.

Reply:
I don't disagree about the general responsibility of things, but it also means that virtually everywhere we call `return unmarshalServerError(res)` we have to place `defer res.Body.Close()` above it.  and the one thing that bugs me there is we still aren't closing that response body on successful `Export`.

happy to revert this and be explicit with `defer res.Body.Close()` in all the `unmarshalServerError` spots.  the other change i'm including is not exporting `UnmarshalServerError` from the `realm` package, not sure why I needed that before.

Reply:
**UPDATE:** took `defer res.Body.Close()` out of the helper, and made sure each client api call was properly calling `Close()` whenever body reads were attempted.  should be all set here